### ISSUE TITLE
Improve enum values for resources

### ISSUE THREAD

**Issue by buehler** [zitadel/terraform-provider-zitadel#26]
When using the ZITADEL terraform provider, some fields are unnecessary complex.

For example, when creating an OIDC application, the enum for "auth_method_type" is neither documented, nor validated.

Instead of just using the Go enum const value, I suggest using custom values to make them clear and be able to document them properly:

Possible values for `auth_method_type`: "none", "jwt_profile", "basic", ... instead of "OIDC_AUTH_METHOD_TYPE_NONE", "OIDC_AUTH_METHOD_TYPE_BASIC", ...

I found a somewhat enum like definition for terraform variables:
```tf
variable "input_parameter" {
  type    = string
  default = "a"

  validation {
    condition     = contains(["a", "b", "c"], var.input_parameter)
    error_message = "Allowed values for input_parameter are \"a\", \"b\", or \"c\"."
  }
}
```

**Comment by stebenz** [comment]
To have it 1:1 reflect the ZITADEL API we are currently using the defined enums, could be an improvement in the future to use easier defined values, good point.
But the values should be validated in the resource code with the defined enum list, do you have an example where this is not happening?

**Comment by buehler** [comment]
It is not an issue with validation, but with autocompletion. As a user, I have no idea which values are possible since they are not documented. If the description on the terraform site includes a list of possible values with a small description, the problem is mitigated.

**Comment by buehler** [comment]
As an example: right now I try to find the correct value to acc a zitadel_org_member. 
```tf
resource "zitadel_org_member" "api-access" {
  org_id  = local.zitadel_org_id
  user_id = zitadel_machine_user.api-access.id
  roles   = ["ORG_OWNER"]
}
```

While the example includes ORG_OWNER, I have no idea which other values are possible. Right now I need to go through the proto files of zitadel (or the code) to find the possible values for this roles array.

**Comment by stebenz** [comment]
supported values for enums included in the documentation with #34

**Comment by stebenz** [comment]
included in release 1.0.0-alpha.10 so closing issue
